python/plugins/processing/algs/qgis/StatisticsByCategories.py

```diff
@@ -90,7 +91,10 @@ def processAlgorithm(self, parameters, context, feedback):
             feedback.setProgress(int(current * total))
             attrs = feat.attributes()
             try:
-                value = float(attrs[value_field_index])
+                if value_field_index > 0:
```

Comment:
shouldn't this be `>= 0`?

Reply:
Yes, very sorry about that.  I did the testing on another computer...